sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/logging/KalixLogbackConfigurator.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.javasdk.impl.logging
+
+import java.io.File
+
+import ch.qos.logback.classic.LoggerContext
+import ch.qos.logback.classic.spi.Configurator
+import ch.qos.logback.core.spi.ContextAwareBase
+
+class KalixLogbackConfigurator extends ContextAwareBase with Configurator {
+
+  private val devModeLogbackFile = "src/main/resources/logback-dev-mode.xml"
+  // this file path only exist in dev-mode. Once packaged we don't have src/main/resources anymore
+  private val devModeLogback = new File(sys.props("user.dir"), devModeLogbackFile)
+
+  // not really related to logging, but we can disable it here as well
+  System.setProperty("spring.main.banner-mode", "off")
+
+  if (devModeLogback.exists) {
```

Comment:
what happens when running tests? not using logback-test.xml any more, or are these system properties set by the build tool when running tests?

Reply:
Hmm, I think that might be an issue. 

If I recall correctly the ordering is: 

* check system property
* if absent, `logback-test.xml`
* if absent use `logback.xml`

I will verify this.

Reply:
yep, that's the case. If `-Dlogback.configurationFile` is set, it gets precedence.

And we have been doing so for a while. 

Before, we were passing `-Dlogback.configurationFile` in the pom.xml/build.sbt. Now we are passing it from inside the maven/sbt plugins. In this PR, we move it to the logback configurator. 

So, `src/main/resources/logback-dev-mode.xml` is always getting precedence over `src/test/resources/logback-test.xml`.

We can fix it by using `src/test/resources/logback-test.xml`, if present, in dev-mode, and falling back to `src/main/resources/logback-dev-mode.xml` if not present. 

Next, we can update our templates to not include `logback-dev-mode.xml` anymore, but always include a `logback-test.xml`. Arguably, dev-mode is also a kind of test run.

Reply:
an alternative is to only configure `logback-dev-mode.xml` if `logback-test.xml` is not in the classpath. If `logback-test.xml` is in the classpath, it means that we are running the tests and therefore we should favour `logback-test.xml`.

Reply:
yeah, could be confusing with many different files and unclear which is used when.

note that logback-test.xml is probably not in the classpath when running the app because it is typically placed in `src/test/resources`. At least it is not included when running from a packaged image. Could be included if running from the IDE (if we support that).

> always include a logback-test.xml. Arguably, dev-mode is also a kind of test run

still somewhat strange to place that in main/resources

logback-test.xml is probably well know and somewhat strange if we wouldn't support it, but maybe too much to include it in our templates. Then we could prefer logback-test in dev mode (if they have added it themselves) otherwise we use logback-dev-mode.xml

Reply:
Actually, our templates include a `src/test/resources/logback-test.xml` and `src/main/resources/logback-dev-mode.xml`. 

Current PR is now looking for `logback-test.xml` in classpath, therefore only present when running the tests. 

Otherwise, it will pick `src/main/resources/logback-dev-mode.xml`, but not from the classpath, but directly fro the dir. Therefore only present if running on local machine and not when running a packaged jar. 

The current code is somewhat future-proof in the sense that we can later decide to only provide `logback-test.xml` and drop `logback-dev-mode.xml`. Then, when running application locally, we would point `logback.configurationFile` to `src/test/resources/logback-test.xml`.